include/zephyr/net/net_event.h

```diff
@@ -53,6 +53,8 @@ enum net_event_if_cmd {
 #define NET_EVENT_IF_ADMIN_UP				\
 	(_NET_EVENT_IF_BASE | NET_EVENT_IF_CMD_ADMIN_UP)
 
+#define NET_EVENT_IF_HOSTNAME_CHANGED			\
```

Comment:
To reviewers - as the hostname identifies a machine and not an interface I wondered if it is suitable to add new event layer with some generic category like `NET_EVENT_MISC_*` or `NET_EVENT_GENERAL_*`

Reply:
After reconsideration - I would put the hostname change event in NET_EVENT_L4_CMD* space as in the network context its used on the application layer (e.g. name resolving).